src/main/java/com/gtnewhorizon/gtnhlib/mixins/early/MixinGuiScreen.java

```diff
@@ -0,0 +1,178 @@
+package com.gtnewhorizon.gtnhlib.mixins.early;
+
+import static com.gtnewhorizon.gtnhlib.client.event.RenderTooltipEvent.*;
+
+import java.util.List;
+
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.entity.RenderItem;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.common.MinecraftForge;
+
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL12;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import com.gtnewhorizon.gtnhlib.client.event.RenderTooltipEvent;
+
+// This mixin must run before AppleCore's GuiScreenMixin because we're using an @Overwrite
+@Mixin(priority = 999, value = GuiScreen.class)
+public class MixinGuiScreen extends Gui {
+
+    @Unique
+    private ItemStack gtnhlib$currentStack;
+
+    @Shadow
+    protected static RenderItem itemRender;
+    @Shadow
+    public int width;
+    @Shadow
+    public int height;
+
+    @Inject(at = @At("HEAD"), method = "renderToolTip")
+    private void preRenderToolTip(ItemStack itemIn, int x, int y, CallbackInfo ci) {
+        this.gtnhlib$currentStack = itemIn;
+    }
+
+    @Inject(at = @At("TAIL"), method = "renderToolTip")
+    private void postRenderToolTip(CallbackInfo ci) {
+        this.gtnhlib$currentStack = null;
+    }
+
+    /**
+     * @author glowredman
+     * @reason Add RenderTooltipEvent
+     */
+    @Overwrite(remap = false)
+    protected void drawHoveringText(List<String> textLines, int mouseX, int mouseY, FontRenderer font) {
```

Comment:
I chose to `@Overwrite` because otherwise I would've needed multiple other mixins (for posting the event, for the alternative renderer and for allowing the background to have a gradient). Combining that with the fragility of AppleCore's mixin would've been nearly impossible. I would've needed to switch to ASM at that point.